### PR TITLE
Added Docker development process.

### LINKED ISSUE #72: For developers, Dockerize the application
Dockerizing the application will make it easier for new developers to get started on the project.

**Note:** this is only for development. Creating a Docker process for production release is outside the scope of this issue.

### PR DESCRIPTION
**Pull request by BenOvermyer**
# How to Test

Follow the instructions in the modified README.md. Every step should work as advertised.

This resolves #72.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+FROM ruby:2.3.3
+
+RUN apt-get update -qq
+RUN apt-get install -y \
+    build-essential \
+    libpq-dev \
+    nodejs \
+    sqlite3
+
+RUN mkdir /app
```
Comment (mshappe) [inline Dockerfile]: Let's add
```
VOLUME /app
```
as well. I think it's not strictly necessary, but it helps to make the point that we expect a mount-point there.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+FROM ruby:2.3.3
```
Comment (mshappe) [inline Dockerfile]: When I pull and test all this, I may also update the project to 2.3.5. Nothing you need to do; just letting you know I might submit my own patch on top of this.

File: README.md [hunk 3]
```diff
@@ -29,3 +29,47 @@ Michael Scott Shappe
...
+From the project directory, follow these directions.
+
+First, copy the `config/database.docker.yml` file to `config/database.yml`. This will give Docker the configuration it needs to talk to the databases.
+
+Next, run `docker-compose up`. The first time you run this, it will build the ConOnRails application's Docker image. Future runs will not need to do this and will be much faster.
+
+The third step is to prepare the database. Run these three commands in order:
+
+```
+docker-compose run web rake db:migrate
```
Comment (mshappe) [inline README.md]: I think they also will need to do `db:create`

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+version: '3'
+services:
+  db:
+    image: postgres
```
Comment (mshappe) [inline docker-compose.yml]: You're not giving it its own persistent volume. When the container terminates for any reason, the database will evaporate.

File: README.md (hunk 3 above)
Comment (BenOvermyer) [reply README.md]: No, because the postgres image takes care of that.

File: Dockerfile (hunk 1 above)
Comment (BenOvermyer) [reply Dockerfile]: docker-compose takes care of this.

File: Dockerfile (hunk 2 above)
Comment (BenOvermyer) [reply Dockerfile]: Sure thing.

File: docker-compose.yml (hunk 4 above)
Comment (BenOvermyer) [reply docker-compose.yml]: This was intentional. `docker-compose stop` and `docker-compose down` having different behavior offers more granular options.

**Comment by mshappe** [comment]
Double check the 2.3.5 update (I did run it locally), then feel free to merge!

File: .travis.yml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 language: ruby
 cache: bundler
 rvm:
-  - 2.3.3
+  - 2.3.5
```
Comment (mshappe) [inline .travis.yml]: 👍

**Pull request merged**